src/datastore/pg-write-store.ts

```diff
@@ -0,0 +1,2280 @@
+        await this.refreshMaterializedView(sql, 'chain_tip');
+        const deletedMempoolTxs = await this.deleteGarbageCollectedMempoolTxs(sql);
+        if (deletedMempoolTxs.deletedTxs.length > 0) {
+          logger.verbose(`Garbage collected ${deletedMempoolTxs.deletedTxs.length} mempool txs`);
+        }
+
+        const tokenContractDeployments = data.txs
+          .filter(entry => entry.tx.type_id === DbTxTypeId.SmartContract)
+          .filter(entry => entry.tx.status === DbTxStatus.Success)
+          .filter(entry => entry.smartContracts[0].abi && entry.smartContracts[0].abi !== 'null')
+          .map(entry => {
+            const smartContract = entry.smartContracts[0];
+            const contractAbi: ClarityAbi = JSON.parse(smartContract.abi as string);
+            const queueEntry: DbTokenMetadataQueueEntry = {
+              queueId: -1,
+              txId: entry.tx.tx_id,
+              contractId: smartContract.contract_id,
+              contractAbi: contractAbi,
+              blockHeight: entry.tx.block_height,
+              processed: false,
+            };
+            return queueEntry;
+          })
+          .filter(entry => isProcessableTokenMetadata(entry.contractAbi));
+        for (const pendingQueueEntry of tokenContractDeployments) {
+          const queueEntry = await this.updateTokenMetadataQueue(sql, pendingQueueEntry);
+          tokenMetadataQueueEntries.push(queueEntry);
+        }
+      }
+    });
+
+    // Skip sending `PgNotifier` updates altogether if we're in the genesis block since this block is the
+    // event replay of the v1 blockchain.
+    if ((data.block.block_height > 1 || !isProdEnv) && this.notifier) {
+      await this.notifier.sendBlock({ blockHash: data.block.block_hash });
+      for (const tx of data.txs) {
+        await this.notifier.sendTx({ txId: tx.tx.tx_id });
+      }
+      await this.emitAddressTxUpdates(data.txs);
+      for (const tokenMetadataQueueEntry of tokenMetadataQueueEntries) {
+        await this.notifier.sendTokenMetadata({ queueId: tokenMetadataQueueEntry.queueId });
+      }
+    }
+  }
+
+  async updateMinerReward(sql: PgSqlClient, minerReward: DbMinerReward): Promise<number> {
+    const values: MinerRewardInsertValues = {
+      block_hash: minerReward.block_hash,
+      index_block_hash: minerReward.index_block_hash,
+      from_index_block_hash: minerReward.from_index_block_hash,
+      mature_block_height: minerReward.mature_block_height,
+      canonical: minerReward.canonical,
+      recipient: minerReward.recipient,
+      coinbase_amount: minerReward.coinbase_amount.toString(),
+      tx_fees_anchored: minerReward.tx_fees_anchored.toString(),
+      tx_fees_streamed_confirmed: minerReward.tx_fees_streamed_confirmed.toString(),
+      tx_fees_streamed_produced: minerReward.tx_fees_streamed_produced.toString(),
+    };
+    const result = await sql`
+      INSERT INTO miner_rewards ${sql(values)}
+      RETURNING *
```

Comment:
There's a lot of instances of using `RETURNING *` when only the row count is needed. This can be much more efficient by removing `RETURNING *` and using `result.count` instead of `result.length`.

Reply:
Great catch, thanks. Must've missed that from the docs (or it isn't there)